pkg/verifier/notation/revocationfactory.go

```diff
@@ -0,0 +1,32 @@
+// Copyright The Ratify Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package notation
+
+import (
+	"net/http"
+
+	"github.com/notaryproject/notation-core-go/revocation"
+	corecrl "github.com/notaryproject/notation-core-go/revocation/crl"
+)
+
+type revocationFactory interface {
+	// NewFetcher returns a new fetcher instance
+	NewFetcher(*http.Client) (*corecrl.HTTPFetcher, error)
+
+	// NewFileCache returns a new file cache instance
+	NewFileCache(string) (corecrl.Cache, error)
```

Comment:
we can rename it to `NewCache` as we don't assume the cache is always file-based, correct me if I'm wrong.

Reply:
I'm wondering if we can just expose NewFetcher API instead of `NewFetcher` and `NewFileCache`. I feel the `NewFetcher` API could handles internal construction of cache and fetcher to facilitate creating a new fetcher.

Reply:
Good point, we can have the constructor of cache inside the `NewFetcher` as discussed